14/README.md

```diff
@@ -0,0 +1,173 @@
+- OFF: Nothing is logged
+- FATAL: A fatal error occurred. The process is about to abort
+- ERROR: An (unrecoverable) error occurred but the process is still alive
+- WARN: An error occurred but could be recovered locally
+- INFO: Provides information on important actions performed
+- DEBUG: Generates detailed information describing the internal behavior of a device
+
+Levels SHALL be ordered the following way:  
+```DEBUG < INFO < WARN < ERROR < FATAL < OFF```  
+For a given device, a level SHALL be said to be enabled if it is greater or equal to the logging level assigned to this device.
+In other words, any logging request which level is lower than the device's logging level is ignored.
+
+The all the Device's Logging Targets SHALL share the same device logging level.
+
+### C++ API
+TLS SHALL provide the following C++ macros for generating messages.
+
+There SHALL be two forms, printf-like and stream-like.
+
+printf-like: ```LOG_DEBUG(("Msg#%d - Hello world", i++));```
+
+stream-like: ```DEBUG_STREAM << "Msg#" << i++ << "- Hello world" << endl;```
+
+These two logging requests are equivalent. Note the double parenthesis in the printf version.
+
+The examples just given are for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+TLS SHALL provide a means to log in the name of a device.  This SHALL be provided when the Device must is a member of a class inheriting from the ```Tango::LogAdapter``` class.
+
+### Python API
+TLS SHALL provide the folliwng Python forms of writing log messages:
+
+basic logging: ```self.debug_stream("read_voltage(%s, %d)", self.host, self.port)```
+
+logging with print: ```print >>self.debug_info, "read voltage attribute"```
+
+logging with decorators: ```@PyTango.DebugIt()```
+
+The examples just given are for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+
+### Java API
+TLS SHALL provide configuration to allow Java to log messages in the following manner:   
+```get_logger().debug(Initializing device  + get_name());```   
+The example just given is for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+
+### Log Consumer
+When a Device Target is specified as "device", this SHALL designate a Device Server which is implementing the Log Consumer interface.
+To be a Log Consumer, a Tango Device SHALL implement just one Tango Command:   
+```void log (Tango::DevVarStringArray details)```.   
+
+The array of strings composing the argument SHALL be interpreted with the following meanings:
+- details[0] : the timestamp in millisecond since epoch (01.01.1970)
+- details[1] : the log level
+- details[2] : the log source (i.e. device name)
+- details[3] : the log message
+- details[4] : the log NDC (contextual info) - Not used but reserved
+- details[5] : the thread identifier (i.e. the thread from which the log request comes from)
+
+
+### GUI for viewing logs
```

Comment:
This I would remove, as GUI is not part of the kernel library

Or just say "TLS client MAY be a GUI app. Tango V9 has LogViewer" and put a link to LogViewer repo or docs